pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -153,18 +156,20 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 	}
 
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "frequency_plan_ids") {
-		fps := make([]*frequencyplans.FrequencyPlan, 0, len(req.Gateway.FrequencyPlanIds))
-		var err error
-		for _, fpID := range req.Gateway.FrequencyPlanIds {
-			var fp *frequencyplans.FrequencyPlan
-			fp, err = s.frequencyPlansStore.GetByID(fpID)
-			if err != nil {
-				break
+		fpSlice := make([]*frequencyplans.FrequencyPlan, 0, len(req.Gateway.FrequencyPlanIds))
+		fps, err := s.frequencyPlansStore(ctx)
+		if err == nil {
```

Comment:
Propagate the error here. This is not the kind of failure that should be hidden (since it shouldn't occur at all).